tests/onnxruntime/test_modeling.py

```diff
@@ -61,6 +61,31 @@ def test_load_model_from_hub(self):
         model = ORTModel.from_pretrained(self.ONNX_MODEL_ID)
         self.assertIsInstance(model.model, onnxruntime.capi.onnxruntime_inference_collection.InferenceSession)
         self.assertIsInstance(model.config, PretrainedConfig)
+        self.assertListEqual(model.providers, ["CPUExecutionProvider"])
+        self.assertEqual(model.device, torch.device("cpu"))
+
+    def test_load_model_cuda_provider_list(self):
+        model = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider=["CUDAExecutionProvider"])
+        self.assertListEqual(model.providers, ["CUDAExecutionProvider", "CPUExecutionProvider"])
+        self.assertEqual(model.device, torch.device("cuda"))
+
+    def test_load_model_cuda_provider_str(self):
+        model = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider="CUDAExecutionProvider")
+        self.assertListEqual(model.providers, ["CUDAExecutionProvider", "CPUExecutionProvider"])
+        self.assertEqual(model.device, torch.device("cuda"))
+
+    def test_load_model_several_providers(self):
+        model = ORTModel.from_pretrained(
+            self.ONNX_MODEL_ID, provider=["CUDAExecutionProvider", "TensorrtExecutionProvider"]
+        )
+        self.assertListEqual(
+            model.providers, ["CUDAExecutionProvider", "TensorrtExecutionProvider", "CPUExecutionProvider"]
```

Comment:
I don't understand this one? should `model.provider` return the value which provider is used for inference?

Reply:
Yes it is confusing. ONNX Runtime adds the `CPUExecutionProvider` as a default if it is not provided, see: https://github.com/microsoft/onnxruntime/blob/0d6cbc6e57c2ecd0722cefaa87d300f301864e3b/onnxruntime/core/session/inference_session.cc#L1280-L1287